src/reanimated2/Hooks.js

```diff
@@ -293,12 +320,28 @@ export function useAnimatedStyle(updater) {
 // when you need styles to animated you should always use useAS
 export const useAnimatedProps = useAnimatedStyle;
 
-export function useDerivedValue(processor) {
+export function useDerivedValue(processor, dependencies = []) {
   const initRef = useRef(null);
-  if (initRef.current === null) {
+
+  if (dependencies === null) {
```

Comment:
for now the default value for `dependencies` is `[]` so if this argument is not passed the worklet reconstruction mechanism is not listening on anything. If `null` is passed it is listening on any change so it does not work exactly like the other hooks such as `useEffect`. This is a temporal solution, just wanted to avoid reconstructing every worklet by default, but this needs a discusion. The same refers to `useAnimatedStyle`